tst/EnergyPlus/unit/InputProcessor.unit.cc

```diff
@@ -74,6 +74,25 @@ namespace EnergyPlus {
 
 		}
 
+		TEST_F(InputProcessorFixture, MisleadingIDDWarningTest)
+		{
+			bool errors_found = false;
+
+			std::string const idf_contents = delimited_string({
+				"Timestep, 4;"
+			});
+
+			ASSERT_FALSE( process_idf( idf_contents, errors_found ) );
+			EXPECT_TRUE( compare_err_stream( "", true ) );
+
+			std::string saveMatchVersion = DataStringGlobals::MatchVersion;
+			DataStringGlobals::MatchVersion = "2.0";
+			ASSERT_FALSE(process_idf(idf_contents, errors_found ));
+			EXPECT_TRUE( compare_err_stream( "   ** Severe  ** IP: Possible incorrect IDD File\r\n   **   ~~~   ** " + DataStringGlobals::IDDVerString +" not the same as expected =\"2.0\"\r\n", true ) );
```

Comment:
@mjwitte Ok after looking over the code, I remember why I replicated `ProcessInput()`. That function does a lot of I/O and it would take a decent amount of work to refactor it for unit testing. I could do it but I don't think it is worth it for this PR.

My suggestion is to make the `delimited_string()` change and call this good, even with it testing `process_idf()`. This is a rarely hit error message and we know that process_idf() has the same functionality now.

RE: your suggestion to refactor to checkIDDversion. That doesn't really work either with the current way I have the process_idf() version of that code. I have EXPECT assertions within the unit test version, while the normal version has ShowContinueError.